src/main/java/edu/kit/datamanager/pit/web/impl/TypingRESTResourceImpl.java

```diff
@@ -553,9 +577,18 @@ public ResponseEntity<PIDRecord> getRecord(
         if (applicationProps.getStorageStrategy().storesResolved()) {
             storeLocally(pid, false);
         }
+        this.saveToElastic(rec);
         return ResponseEntity.ok().body(rec);
     }
 
+    private void saveToElastic(PIDRecord rec) {
+        this.elastic.ifPresent(
+            database -> database.save(

```

Comment:
Where does this database variable come from? Just out of curiosity as I don't know this construct.

Reply:
this.elastic is an `Optional<PidRecordElasticRepository>`. `database` is the database repo inside, assuring it is not null. It is shorthand for something like this:

```java
if (this.elastic.isPresent()) {
  PidRecordElasticRepository database = this.elastic.get();
  database.save(...)
}
```

The cool thing about it (except it is shorter) is, that I do not have to manually check things. It is either there and happens, or it is not there and the step will be skipped. It is needed as depending on the configuration, the database repo might be there or not.

Reply:
I didn't say it is bad, just that I didn't know it. ;-) 

Resolved.